gliner/training/trainer.py

```
@@ -279,12 +306,8 @@ def create_optimizer(self):
                 },
             ]
 
-        # Works across v4/v5
-        if hasattr(transformers.Trainer, "get_optimizer_cls_and_kwargs"):
-            optimizer_cls, optimizer_kwargs = transformers.Trainer.get_optimizer_cls_and_kwargs(self.args)
-        else:
-            # very old fallback
-            optimizer_cls, optimizer_kwargs = super().get_optimizer_cls_and_kwargs(self.args)
+        # Works across v4/v5 – static method on Trainer
```

Ruff (0.15.2)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)